feat: add chat input history with arrow key navigation

planger opened the pull request:
#### What it does

Implements prompt history functionality for AI chat input widgets:
  - Add ChatInputHistoryService for persistent prompt storage and retrieval
  - Add ChatInputHistoryContribution with up/down arrow key navigation
  - Integrate history navigation into AIChatInputWidget with context keys
  - Support shared history across chat instances with deduplication
  - Configure enablePromptHistory option in input configuration

#### How to test

Run a couple of prompts in the AI chat and then try navigating through your prompt history with the cursor up/down keys.

#### Follow-ups

<!-- Please list potential follow-up work, including known issues, possible future work, identified technical debt, and potentially introduced technical debt. If the PR introduces technical debt, specify the reason why this is acceptable. Please create tickets and link them here. Please use the label "technical debt" for new issues when it applies. -->

#### Breaking changes

- [ ] This PR introduces breaking changes and requires careful review. If yes, the breaking changes section in the [changelog](https://github.com/eclipse-theia/theia/blob/master/CHANGELOG.md) has been updated.

#### Attribution

Contributed on behalf of STMicroelectronics

#### Review checklist

- [X] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

ndoschek reviewed (commented):
Hi @planger,
Thanks a lot, this is a great improvement :tada:!

I haven't looked at it in detail yet, but I have one small request:
To indicate to the user that history is available, we should use the same approach as the search textarea, which inserts double arrows to show that history browsing is possible once the input element is focused (see below).

What do you think? Thanks in advance!

[search-text-area-placeholder.webm](https://github.com/user-attachments/assets/8e953f2a-f281-439c-9262-ec77f7f3c3e4)

<img width="383" height="105" alt="image" src="https://github.com/user-attachments/assets/eecc7b19-6511-4e8e-acee-3c290bd3d691" />

planger commented:
@ndoschek Absolutely! Good idea. I've added the indicator in the placeholder. Also I've added a command to clear the prompt history, in case someone accidentally pasted something they want to get rid of.

planger commented on packages/ai-chat-ui/src/browser/chat-input-history-contribution.ts:
```diff
@@ -0,0 +1,150 @@
+// *****************************************************************************
+// Copyright (C) 2025 EclipseSource GmbH.
```
Nice catch, thank you!

ndoschek commented:
Hi @planger, I just discovered a small regression related to this new feature. I've documented the issue here: https://github.com/eclipse-theia/theia/issues/16222
Would you mind taking a look when you get a chance? Thanks!
